src/pkcs11/framework-pkcs15.c

```diff
@@ -5552,7 +5552,13 @@ pkcs15_skey_encrypt(struct sc_pkcs11_session *session, void *obj,
 	   can be an init operation or final operation..*/
 
 	if (skey && !(skey->info->usage & SC_PKCS15_PRKEY_USAGE_ENCRYPT))
+		skey = NULL;
+
+	/* Please read comments in pkcs15_skey_unwrap() and pkcs15_skey_wrap() */
+
+	if (skey == NULL)
 		return CKR_KEY_FUNCTION_NOT_PERMITTED;
```

Comment:
Isn't this doing the same as the following, with less obfuscation?

```suggestion
	if (!skey || !(skey->info->usage & SC_PKCS15_PRKEY_USAGE_ENCRYPT))
		return CKR_KEY_FUNCTION_NOT_PERMITTED;
```

I am also not sure what the comment is trying to say here, whether it is related to the change you made or to something else in this function.

Reply:
We need to prevent dereferencingf NULL pointer (in case `skey` is NULL). But, there is already similar code in another functions:
````
	/* Check whether this key supports unwrap */
	if (skey && !(skey->info->usage & SC_PKCS15_PRKEY_USAGE_UNWRAP))
		skey = NULL;

	/* TODO: should we look for a compatible key automatically? prv_next not implemented yet. */
	/* skey = skey->prv_next; */

	if (skey == NULL)
		return CKR_KEY_FUNCTION_NOT_PERMITTED;
````
Either we will use the current PR, which we will update in the future (there will be a search for another possible key), or I will change this PR to just a simple test like you suggested.

Reply:
If the comment between the statements will be as in the other functions, I am ok with the code as it is. 

But the current comment really does not help understanding why the statement is broken into separate blocks without reading through comments in other places.